src/react/hooks/use-typing.ts

```diff
@@ -0,0 +1,90 @@
+import { Typing, TypingListener } from '@ably/chat';
+import { useCallback, useEffect } from 'react';
+
+import { ChatStatusResponse } from '../chat-status-response.js';
+import { Listenable } from '../listenable.js';
+import { StatusParams } from '../status-params.js';
+import { useChatConnection } from './use-chat-connection.js';
+import { useRoom } from './use-room.js';
+
+/**
+ * The options for the {@link useTyping} hook.
+ */
+export interface TypingParams extends StatusParams, Listenable<TypingListener> {
+  /**
+   * A listener that will be called whenever a typing event is sent to the room.
+   */
+  listener?: TypingListener;
+}
+
+export interface UseTypingResponse extends ChatStatusResponse {
+  /**
+   * A shortcut to the {@link Typing.start} method.
+   */
+  readonly start: Typing['start'];
+
+  /**
+   * A shortcut to the {@link Typing.stop} method.
+   */
+  readonly stop: Typing['stop'];
+
+  /**
+   * A shortcut to the {@link Typing.get} method.
+   */
+  readonly currentlyTyping: Typing['get'];
+
+  /**
+   * Provides access to the underlying {@link Typing} instance of the room.
+   */
+  readonly typingIndicators: Typing;
+}
+
+/**
+ * A hook that provides access to the {@link Typing} instance in the room.
+ * It will use the instance belonging to the room in the nearest {@link RoomProvider} in the component tree.
+ *
+ * @param params - Allows the registering of optional callbacks.
+ * @returns UseTypingResponse - An object containing the {@link Typing} instance and methods to interact with it.
+ */
+export const useTyping = (params?: TypingParams): UseTypingResponse => {
+  const { currentStatus: connectionStatus, error: connectionError } = useChatConnection({
+    onStatusChange: params?.onConnectionStatusChange,
+  });
+  const { room, roomError, roomStatus } = useRoom({
+    onStatusChange: params?.onRoomStatusChange,
+  });
+
+  // if provided, subscribes the user provided onDiscontinuity listener
+  useEffect(() => {
+    if (!params?.onDiscontinuity) return;
+    const { off } = room.typing.onDiscontinuity(params.onDiscontinuity);
+    return () => {
+      off();
+    };
+  }, [room, params]);
+
+  // if provided, subscribe the user provided listener to TypingEvents
+  useEffect(() => {
+    if (!params?.listener) return;
+    const { unsubscribe } = room.typing.subscribe(params.listener);
+    return () => {
+      unsubscribe();
+    };
+  }, [room, params]);
+
+  // memoize the methods to avoid re-renders, and ensure the same instance is used
+  const currentlyTyping = useCallback(() => room.typing.get(), [room.typing]);
+  const start = useCallback(() => room.typing.start(), [room.typing]);
+  const stop = useCallback(() => room.typing.stop(), [room.typing]);
+
+  return {
+    connectionStatus,
+    connectionError,
+    roomStatus,
+    roomError,
+    start,
+    stop,
+    currentlyTyping,
```

Comment:
Should `currentlyTyping` perhaps be a property, that automatically gets updated by some internal listener, similar to what we have with occupancy?

Reply:
It can indeed, I'll add the same throttle capability as the `useOccupancy` hook :)